src/object/srv_obj_migrate.c

```diff
@@ -1350,20 +1350,19 @@ migrate_fetch_update_bulk(struct migrate_one *mrone, daos_handle_t oh,
 	/* For EC object, if the migration include both extent from parity rebuild
 	 * and extent from replicate rebuild, let rebuild the extent with parity first,
 	 * then extent from replication.
-	 *
-	 * Since the parity shard epoch should be higher or equal to the data shard epoch,
-	 * so let's use the minimum epochs of all parity shards as the update epoch of
-	 * this data shard.
 	 */
-
 	for (i = 0; i < mrone->mo_iods_num_from_parity; i++) {
 		for (j = 0; j < mrone->mo_iods_from_parity[i].iod_nr; j++) {
 			daos_iod_t iod = mrone->mo_iods_from_parity[i];
+			daos_epoch_t fetch_eph;
 
 			iod.iod_nr = 1;
 			iod.iod_recxs = &mrone->mo_iods_from_parity[i].iod_recxs[j];
-			rc = __migrate_fetch_update_bulk(mrone, oh, &iod, 1,
-							 mrone->mo_iods_update_ephs_from_parity[i][j],
+			/* fetch epoch can not be higher than rebuild stable epoch(mo_eph),
+			 * to avoid fetching the inflight(new) data.
+			 */
+			fetch_eph = min(mrone->mo_epoch, mrone->mo_iods_update_ephs_from_parity[i][j]);
```

Comment:
just confirm that the "mrone->mo_epoch" is the rebuild epoch (stable epoch) and also the dsc_obj_list_obj()'s epoch range's epr_hi?
the mo_iods_update_ephs_from_parity is the enumerated ext's epoch?
then mo_iods_update_ephs_from_parity should not be higher than the list_obj's epoch (rebuild stable epoch) right? thx

Reply:
>just confirm that the "mrone->mo_epoch" is the rebuild epoch (stable epoch) and also the dsc_obj_list_obj()'s epoch range's epr_hi?

right.  Enumerate epoch is already limited by the stable epoch. And this patch will limit the epoch by stable epoch as well.   This is actually a regression from recent PR.  Sorry I may miss your question here.